controllers/chat/findOrCreateChannelBySignedSender.js

```diff
@@ -0,0 +1,48 @@
+const {StreamChat} = require('stream-chat');
+const {CHANNEL_TYPE_STRING, CHANNEL_TYPE} = require('../../helpers/constants');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const findOrCreateChannelBySignedSender = async (req, res) => {
+  const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  const {members} = req.body;
+
+  try {
+    if (!members.includes(req.userId)) members.push(req.userId);
+
+    if (members.length < 2) return ErrorResponse.e400(res, 'Members must be more than 1');
+
+    await client.connectUser({id: req.userId}, req.token);
+
+    const channelTypeString =
+      members.length > 2 ? CHANNEL_TYPE_STRING.GROUP : CHANNEL_TYPE_STRING.CHAT;
+    const channelType = members.length > 2 ? CHANNEL_TYPE.GROUP : CHANNEL_TYPE.CHAT;
+
+    const channel = client.channel(channelTypeString, {
+      members
+    });
+
+    const findOrCreateChannel = await channel.create();
+    const channelName = findOrCreateChannel.members.map((member) => member.user.name).join(', ');
+    await channel.updatePartial({
+      set: {
+        channelType,
+        channel_type: channelType,
+        name: channelName
+      }
+    });
+
+    await client.disconnectUser();
+
+    return res.status(200).json(findOrCreateChannel);
+  } catch (error) {
+    await client.disconnectUser();
+    return ErrorResponse.e400(res, error.message);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the catch block, it's better to disconnect the user before returning the error response. This ensures that the user is disconnected even if sending the response fails for some reason.
```diff
-    await client.disconnectUser();
-    return ErrorResponse.e400(res, error.message);
+    try {
+      await client.disconnectUser();
+    } finally {
+      return ErrorResponse.e400(res, error.message);
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->